iphone/Classes/GeolocationModule.m

```diff
@@ -869,7 +869,10 @@ - (void)requestLocationPermissioniOS7:(id)args {
 -(void)requestLocationPermissions:(id)args
 {
     if (![TiUtils isIOS8OrGreater]) {
-        [self requestLocationPermissioniOS7:args];
+        // It is required that delegate is created and permission is presented in main thread.
+        TiThreadPerformOnMainThread(^{
+            [self requestLocationPermissioniOS7:args];
+        }, NO);
```

Comment:
I don't see any dependency to the `TI_USE_KROLL_THREAD`, did you test it with both `<run-on-mainthread/>` and without? Just want to double-check.